internal/cmd/commands/server/server.go

```diff
@@ -133,21 +134,12 @@ func (c *Command) AutocompleteFlags() complete.Flags {
 }
 
 func (c *Command) Run(args []string) int {
-	ctx := context.TODO()
 	c.CombineLogs = c.flagCombineLogs
 
 	if result := c.ParseFlagsAndConfig(args); result > 0 {
 		return result
 	}
 
-	if c.configWrapper != nil {
-		defer func() {
-			if err := c.configWrapper.Finalize(c.Context); err != nil {
```

Comment:
Reviewers: Note the change in context where this is moved to -- it uses context.Background(). This is because if the context is canceled due to Ctrl-C, we still want to attempt to finalize the wrapper.

Future work could include adding a timeout somewhere in the finalize call so it can't hang forever, but this is still safer than using an already-canceled context.